src/interfaces/simulation.ts

```diff
@@ -38,84 +44,145 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     options: SimulationOptions = {}
   ): Promise<TransactionOutcome> {
     const signer = this.ctx.provider.write.getSigner(from);
-    const zapProtocol = PickleJars.includes(toVault) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-    const vaultContract =
-      zapProtocol === ZapProtocol.PICKLE
-        ? new PickleJarContract(toVault, signer)
-        : new YearnVaultContract(toVault, signer);
-
-    const underlyingToken = await vaultContract.token().catch(() => {
-      throw new EthersError("failed to fetch token", EthersError.FAIL_TOKEN_FETCH);
-    });
-    const isZapping = underlyingToken !== sellToken;
-    let forkId: string | undefined;
-    let simulateDeposit: (save: boolean) => Promise<TransactionOutcome>;
 
-    if (isZapping) {
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
+    const vaultContract = this.getVaultContract({ toVault, signer });
 
-      let needsApproving: boolean;
+    const vaultMetadata = await this.yearn.services.meta.vault(toVault);
 
-      if (sellToken === EthAddress) {
-        needsApproving = false;
-      } else {
-        needsApproving = await this.yearn.services.zapper
-          .zapInApprovalState(from, sellToken, zapProtocol)
-          .then((state) => !state.isApproved)
-          .catch(() => {
-            throw new ZapperError("approval state", ZapperError.ZAP_IN_APPROVAL_STATE);
+    // TODO Fetch the token
+    const token = {};
```

Comment:
@xgambitox what would be the best way to get this token?

Reply:
As discussed, went with https://github.com/yearn/yearn-sdk/pull/280/commits/c22a049b7916cb1b59307f9259c0b47bc6b5ccb9 -- 
```